svelte-kit-scss/src/lib/components/auth/Auth.spec.ts

```diff
@@ -0,0 +1,15 @@
+import { render, screen } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import Auth from './Auth.svelte';
+
+describe('Auth Component', () => {
+  it('should render sign in', async () => {
+    render(Auth);
+
+    const signInBtn = screen.getByRole('button');
+    expect(signInBtn.textContent).toEqual('Sign in with GitHub');
```

Comment:
This test is too heavily tied to what the text of the button is, which is an implementation detail. In this case, if we renamed that to "Login with Github" then the test would fail, even though the actual functionality works.

## Suggestion
Rather than testing for equality of the text button's text content, simply check if we have a sign-in button by for example using `data-testid="button-signin"` which would remain constant even if the actual content of the sign-in button changed.

Reply:
using testid is discouraged, and should be used when it's not possible to get a node in a different way. the text in the button is what matters to the user. getByRole is the right choice in this case. you can use the options to make the query precise, and also be more flexible on the text context.
queries are assertions too, it will throw if it finds more than one node or none.
replacing

```js
const signInBtn = screen.getByRole('button');
expect(signInBtn.textContent).toEqual('Sign in with GitHub'); 
```

with

```js
const signInBtn = screen.getByRole('button', { name: /sign in with github/i });
```

should be enough to assert that the button exists, and has more flexibility with the text in it. a change in casing doesn't really matter to the user. It's also resilient to adding more buttons.
Hope it helps.